src/Integration.Vsix/CFamily/CMake/CompilationDatabaseRequest.cs

```diff
@@ -36,16 +36,17 @@ internal class CompilationDatabaseRequest : IRequest
         private readonly CompilationDatabaseEntry databaseEntry;
         private readonly IRulesConfigProtocolFormatter rulesConfigProtocolFormatter;
 
-        public CompilationDatabaseRequest(CompilationDatabaseEntry databaseEntry, RequestContext context)
-            : this(databaseEntry, context, new RulesConfigProtocolFormatter())
+        public CompilationDatabaseRequest(CompilationDatabaseEntry databaseEntry, RequestContext context, IReadOnlyDictionary<string, string> environmentVariables)
```

Comment:
I wonder if this is something that should be on the context.
We discussed that the subprocess already supports env vars, so logically it makes sense for it to be for all request types.

Reply:
It could be, I guess. It would be easy enough to move.

Reply:
On second thoughts, I don't think it should be. The environment variables are calculated by the factory, and form part of the request protocol itself (env vars + binary input message).

Looking at the `RequestContext` again, I don't think it's responsibilities or role are clear - it seems to be doing multiple things.